### DISCUSSION TITLE
Making JPA converters directly available in webauthn4j-spring-security-core

### DISCUSSION THREAD

**Discussion by TGITS** [webauthn4j/webauthn4j-spring-security#1168] [Ideas]
Hello,

Could it be possible to make the JPA converters from `webauthn-spring-security/samples/spa` directly available in `webauthn4j-spring-security-core` ?

The converters under `com.webauthn4j.springframework.security.webauthn.sample.infrastructure.util.jpa.converter` are quite generic and can be useful when using `webauthn4j-spring-security-core` to implement a solution using a relational database.

Will a PR with such a refactoring be welcomed ?

**Comment by ynojima** [comment]
Adding JPA converters requires adding dependency from `webauthn4j-spring-security` to JPA. It is not welcomed by users who doesn't use JPA. 
As far as I know, Spring Security doesn't have converters for a specific ORM too.
Since JPA converters from `webauthn4j-spring-security/samples/spa` is a thin wrapper for converters from webauthn4j, I think it is not so hard for users to add these JPA converters to their projects.
If we really add these JPA converters to webauthn4j-spring-security project, we need to add a separate module named `webauthn4j-spring-security-core-jpa`, but I don't see enough value on it.

**Comment by TGITS** [comment]
Thanks for your detailled answer.
I understand and I should I have thought about it.
You are right, it would create an unwanted dependency and as such it is not a good idea.
You are right it is easy to copy the converter classes but there may be a little problem depending on your related _corporate_ context : the code is under Apache Licence and if you copy or modifiy it, you should keep the licence. Sometimes it is not possible or it is complicated.
